camp/camp-brooklyn/src/main/java/org/apache/brooklyn/camp/brooklyn/spi/dsl/methods/BrooklynDslCommon.java

```diff
@@ -384,24 +445,48 @@ public Object apply(List<Object> input) {
                                     config.put(name, ((DeferredSupplier<?>) value).get());
                                 }
                             }
-                            return create(type, fields, config);
+                            if (factoryMethodName == null) {
+                                return create(clazz, constructorArgs, fields, config);
+                            } else {
+                                return create(clazz, factoryMethodName, factoryMethodArgs, fields, config);
+                            }
                         }
                     }, tasks);
         }
 
-        public static <T> T create(Class<T> type, Map<String,?> fields, Map<String,?> config) {
+        public static <T> T create(Class<T> type, List<?> constructorArgs, Map<String,?> fields, Map<String,?> config) {
             try {
                 T bean;
                 try {
                     bean = (T) TypeCoercions.coerce(fields, type);
                 } catch (ClassCoercionException ex) {
-                    bean = Reflections.invokeConstructorFromArgs(type).get();
+                    bean = Reflections.invokeConstructorFromArgs(type, constructorArgs.toArray()).get();
+                    BeanUtils.populate(bean, fields);
+                }
+                if (bean instanceof Configurable && config.size() > 0) {
+                    ConfigBag configBag = ConfigBag.newInstance(config);
+                    FlagUtils.setFieldsFromFlags(bean, configBag);
+                    FlagUtils.setAllConfigKeys((Configurable) bean, configBag, true);
+                }
+                return bean;
+            } catch (Exception e) {
+                throw Exceptions.propagate(e);
+            }
+        }
+
+        public static Object create(Class<?> type, String factoryMethodName, List<Object> factoryMethodArgs, Map<String,?> fields, Map<String,?> config) {
+            try {
+                Object bean;
+                try {
+                    bean = TypeCoercions.coerce(fields, type);
```

Comment:
Yeah, I don't really know why that was in the previous version either: I can't think of a sensible use-case, and the unit tests don't cover it. I opted to preserve it rather than risk breaking some obscure corner case, but I'd be *very* happy to see it removed.